flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/sessionjob/FlinkSessionJobReconciler.java

```diff
@@ -76,6 +78,24 @@ public void reconcile(FlinkSessionJob flinkSessionJob, Context context) throws E
                 OperatorUtils.getSecondaryResource(
                         flinkSessionJob, context, configManager.getOperatorConfiguration());
 
+        var labels = flinkSessionJob.getMetadata().getLabels();
+        if (labels == null) {
+            labels = new HashMap<>();
+        }
+        if (!flinkSessionJob
+                .getSpec()
+                .getDeploymentName()
+                .equals(labels.get(LABEL_SESSION_CLUSTER))) {
+            labels.put(LABEL_SESSION_CLUSTER, flinkSessionJob.getSpec().getDeploymentName());
+            flinkSessionJob.getMetadata().setLabels(labels);
+            kubernetesClient
```

Comment:
Can't we use an update control for this?

Reply:
The reconciler do not return `UpdateControl` now, If we want to use the update control for it, we should adjust the interface first to return the update control :/ . We have just decide not to do [this](https://issues.apache.org/jira/browse/FLINK-26915) now. I'm not sure whether we need to break it for this PR.

Reply:
Got it, but this is the first use case that we modify something else then the status. We might need to reconsider this again. Updating the CR deep in the stack sounds worse to me.

Reply:
Yes, agree with it. I'm ok to rework the interface to work out it. Also cc @gyfora @wangyang0918

Reply:
We hit some race conditions anyway with the status updates, so it maybe good as is with the java-operator-sdk v2 and we can change it later.

Reply:
As I know we have manually patch the status after https://github.com/apache/flink-kubernetes-operator/pull/199 . So this change do not bring the resourceVersion conflicts, I think. Do you mean keep it as now ?

Reply:
Let's see what @gyfora and @wangyang0918 think/suggest.

Reply:
if you don't explicitly set the resourceversion to `null` before patching you might still get an error.

Reply:
yeah, but I think this operation is retriable